### DISCUSSION TITLE
PBA Faculty for UC Berkely 2023

### DISCUSSION THREAD

**Discussion by nuke-web3** [Polkadot-Blockchain-Academy/pba-content#595] [General]
This post defines the [faculty and staff](https://www.indeed.com/career-advice/career-development/faculty-vs-staff) selections for the Academy's third edition at the University California, Berkeley - July 10th to August 10th 2023.
There are also a few notes on motivations for those new to the team, as well as highlighting a few new and major refactors of content that are in the works.

> Key changes in content and sequencing:
>
> - [Polkadot](#7-Polkadot) to have Shawn add and refactor content focused on extending FRAME into Parachain engineering specific Polkadot content.
> - [Smart Contracts](#4-Smart-Contracts) moved up from last mod to preceding Substrate.
> - [Additional Content in Consideration](#Additional-Content-in-Consideration) has a few tentative ideas including a start for a new module on `Crypto and Econ Applications into Blockchain` (defined below)

---

## 00 Content Support Staff

Non-technical, non-SME contributors to be on site to audit the course and interface with the venue staffing and support faculty in non-content aspects (AV support, messaging / announcements, managing student recourse pages etc.)

1. **Erin Woggon (Parity, PBA)** (UCB req. dates: July 10, 2023 → August 10, 2023)
   - Lead on student facing resources management & communications.
1. **Alicia and Safina (Parity)** (UCB req. dates: July 10, 2023 → August 10, 2023)
   - One of these great people will in the classroom at all times as a logistical TA.
   - Able to support things like AV for lessons, "corralling" students, relaying information to the team.
1. **Nuke (Parity, PBA)** - @nukemandan (UCB req. dates: July 10, 2023 → August 10, 2023)
   - Content production lead: wholistic view of the program and coordinating all faculty throughout the program.
   - Suggests new and refactored content for module leads.
   - Coordinates with team to incorporate guest and social events around the core content timeline and sequencing.
   - Provides guidelines, templates, and examples for all content types for faculty to use.
   - Owns content policies for sharing, grading, etc. that faculty and students are to abide by.
   - Contributes as able to instruct on site and create content.

### Lead Instructor(s):

1. **Erin Woggon (Parity, PBA)** (UCB req. dates: July 10, 2023 → August 10, 2023)
   - On site welcome, works with PBA staff to brief the new students.
1. **TBD** - There is roughly a 1/2 day of content possible, but we start module 1 with @NukeManDan and @naterarmstrong  the 1st day if no compelling content can be sourced.
   Ideas:
   - Focused Rust sort-course with a review of the common issues found in those taking the [Rust Qualifying Exam](https://github.com/Polkadot-Blockchain-Academy/pba-qualifier-exam) and "assignment 0" that covers more adv. rust patterns and syntax used in substrate {a WIP at this time in [this repo](https://github.com/Polkadot-Blockchain-Academy/Rust-Entrance-Exam)}
   - Wallet generation for use at the academy & security practical 101
   - ... other great ideas? Please comment below :)

### Lead Instructor(s):

1. **Nate Armstrong (Alumni, https://eluvio.co/)** - @elv-nate && @naterarmstrong (UCB req. dates: July 10, 2023 → July 16, 2023)
   - Top-of-class in PBA Buenos Aires, also proactively helped students learn in major ways.
   - Studied Cryptography in Uni, and is working as an engineer building and using cryptographic tools in his day-job.

### TA(s):

1. **Nuke (Parity, PBA)** - @nukemandan (UCB req. dates: July 10, 2023 → August 10, 2023)
1. (more if needed, Gav & Nate's call)

### Remote content support:

1. (TBD if needed, none at present)

### Guest Content

1. **_TBD local SME - At this moment, seems quite unlikely to pan out._**
   - The Berkeley Center for Responsible, Decentralized Intelligence (RDI) _may_ help identify a too-good-too-pass-up candidate we are still open to.
   - Berkeley is home to some world-class _cryptographers_ we are in contact with our primary partner there with hopes to get a _practitioner in Web3 applications_ that will not be steering the content into maths, but deliver it as Gav did.
   - [Additional Content in Consideration](#Additional-Content-in-Consideration) on ZKPs and/or perhaps Thor

---

### Lead Instructor(s):

1. **Jonas (W3F)** - @jonasW3F (UCB req. dates: July 12, 2023 → July 16, 2023)
   - Main content developer, owning 1st part of lessons
1. **Alfonso (W3F)** - @AlfonsoCev (UCB req. dates: July 13, 2023 → July 15, 2023)
   - Main content developer, owning 2nd part of lessons

### TA(s):

1. **Andrew Burger (PBA Vendor)** - @coax1d (UCB req. dates: July 12, 2023 → July 22, 2023)
   - Owns econ games end-to-end using exiting W3F tooling and consider using MVP for some aspects of in-house on-chain games with Faisal within PBA3.
   - Co-owns activities and assignment for the econ module with leads

### Remote content support:

1. **Faisal Al-Tameemi (Alumni)** - @FaisalAl-Tameemi (UCB req. dates: None)
   - Developing econ games V2 as a substrate and ink! powered app to be used in a future PBA
   - ~~May develop and deliver a short lesson on the "how and why" of building a gaming platform using substrate, ink, and Polkadot.js or Capi.~~ perhaps in a future wave.

---

### Lead Instructor(s):

1. **Joshy Orndorff (PBA Vendor)** - @JoshOrndorff (UCB req. dates: July 15, 2023 → August 5, 2023)
   - Has done a _fantastic_ job in prior waves and a key part of the faculty all around.
1. **Maciej Żyszkiewicz (Parity)** - @overkillus (UCB req. dates: July 18, 2023 →  July 22, 2023)
   - Crypto and Econ Applications into Blockchain SME to create and lead this new content
 
### TA(s):

1. **Andrew Burger (PBA Vendor)** - @coax1d (UCB req. dates: July 12, 2023 → July 22, 2023)
1. **Nuke (Parity, PBA)** - @nukemandan (UCB req. dates: July 10, 2023 → August 10, 2023)

### Remote content support:

1. **(TBD if needed)**

---

## 4 Smart Contracts (Planned: 2023-07-19 → 2023-07-21)

> Considering sequencing, the module will expand to be more holistic on web3 application design built on top of some consensus system as a platform.
> Tracked planned changes in #586 

### Lead Instructor(s):

1. **Joshy Orndorff (PBA Vendor)** - @JoshOrndorff (UCB req. dates: July 15, 2023 → August 5, 2023)
1. **Lauren Lee (Parity, Alumni)** - @laurenelee (UCB req. dates: July 16, 2023 → July 22, 2023)

### TA(s):

1. **Stephen Shelton (Moonbeam)** - @notlesh (UCB req. dates: July 18, 2023 → July 27, 2023, 2023)
1. **Aaron Bassett (Parity, Alumni)** - @aaronbassett (UCB req. dates:  July 17, 2023 → August 2, 2023)
   - Fantastic public speaker from our DevRel team.

### Remote content support:

1. (TBD if needed, none at present)

---

### Lead Instructor(s):

1. **Kian Paimani (Parity)** - @kianenigma (UCB req. dates: 22nd July → 13th August)
   - Intends to _minimize_ time to prepare and refactor and will lean into support from the rest of the PBA faculty if needed to refactor.
1. **Aaron Bassett (Parity, Alumni)** - @aaronbassett (UCB req. dates:  July 17, 2023 → August 2, 2023)
   - Owns all content refactoring, with Kian to review.

### TA(s):

1. **Joshy Orndorff (PBA Vendor)** - @JoshOrndorff (UCB req. dates: July 15, 2023 → August 5, 2023)
1. **Stephen Shelton (Moonbeam)** - @notlesh (UCB req. dates: July 18, 2023 → July 27, 2023, 2023)

### Remote content support:

1. **Andrew Burger (PBA Vendor)** - @coax1d (UCB req. dates: July 12, 2023 → July 22, 2023)

---

### Lead Instructor(s):

1. **Shawn (PBA Vendor)** - @shawntabrizi (UCB req. dates: July 20, 2023 → August 14, 2023)
   - Tracking issue of work scope: https://github.com/Polkadot-Blockchain-Academy/pba-content/issues/592
   - TBD if he desires a co-lead instructor, as he expressed desire to focus more on Polkadot and XCM rather than FRAME this time around.

### TA(s):

1. **Aaron Bassett (Parity, Alumni)** - @aaronbassett (UCB req. dates: July 17, 2023 → August 2, 2023)
   - Day-job focus is mainly on FRAME, and will be "learning the ropes" with the desired outcome to take lead in future PBAs for this mod.
1. **Joshy Orndorff (PBA Vendor)** - @JoshOrndorff (UCB req. dates: July 17, 2023 → August 5, 2023)
1. **Stephen Shelton (Moonbeam)** - @notlesh (UCB req. dates: July 18, 2023 → July 27, 2023, 2023)
### Remote content support:

1. **(TBD if needed, none at present)**

---

## 7 Polkadot (Planned: 2023-08-02 → 2023-08-04)

> Based on feedback and the demands from students and staff, there are two "branches" of this module that are needed to speak to:
>
> - Polkadot's architecture and protocols (core engineering)
> - Using Polkadot as a platform (parachain engineering)
>
> In PBA3 we will refactor to better deliver these two somewhat separate topics.
> We also will stage the content such that latter we are able to more cleanly split it into separate tracks targeting core vs. parachain engineers.

### Lead Instructor(s):

1. **Bradley Olson (Parity, Alumni)** - @BradleyOlson64 (UCB req. dates: July 24, 2023 → August 5, 2023)
   - To co-own module with focus on _Polkadot architecture and protocols_.
1. **Shawn (PBA Vendor)** - @shawntabrizi (UCB req. dates: July 20, 2023 → August 14, 2023)
   - To co-own module with focus on _parachain engineering needs_.
1. **Sophia Gold (Parity)** - @Sophia-Gold (UCB req. dates: August 2, 2023 → August 4, 2023)
   - To coordinate with Shawn an Bradely on leading presentations as they see fit in this module.
  
### TA(s):

1. **Joshy Orndorff (PBA Vendor)** - @JoshOrndorff (UCB req. dates: July 17, 2023 → August 5, 2023)
   - Responsible for the technical aspects of the parachain protocol ([per this thread](https://github.com/Polkadot-Blockchain-Academy/pba-content/discussions/595#discussioncomment-5767218))
 
### Remote content support:

These fine folks may also be called on to deliver content at UCB, as mod team deems fit.

1. **Gabriele Miotti (Alumni)** - @gabriele-0201 (async only)
   - Support on Polkadot core protocols content side, should be included where possible.  
   - High performer in PBA2 and very interested in core engineering work inside and outside PBA!
1. **{Rob H (Parity-async)}** @rphmeier (UCB req. dates: TBD)
   - Not owning any content formally unless he requests to and is on site.
   - Planning on a public facing social event for him to speak at a minimum if he is on site anytime for the PBA.

---

### Lead Instructor(s):

1. **Gav (Parity, W3F)** - @gavofyork (UCB req. dates: 2023-08-07 → 2023-08-09)
   - Will review present content and develop outline to refactor/refine as he sees fit.
   - Craft content directly, or delegate out what work needs to be done to TAs and others.
1. **Keith Yeung (Parity)** - @KiChjang (UCB req. dates: 2023-08-01 → 2023-08-10)
   - Owns content dev or further delegation required to develop content alongside Gav's vision for the refactor.
   - Co-owns lessons and activities as Gav sees fit.

### TA(s):

1. **Francisco Aguirre (Parity, Alumni)** @franciscoaguirre (UCB req. August  1 → August 10)
   - Working on XCM team presently, PBA1 alumni.
1. **(TBD - Parachain partners?)**

### Remote content support:

1. **Alejandro Martinez (Parity)** - @al3mart
   - Will be involved in async (only) content for many mods, XCM in particular, scope of work in-person is yet to be defined.
   - Most interested in XCM related mod work.
   - Lots of practical know-how that could be very useful as a TA in office hours crafting and delivering hands-on material.
1. **(TBD - Parachain partners like Gabe from InvArch and Chris from OAK ?)**

### Guest Content

1. **Chris (CTO @ OAK network)** - @chrisli30
   - Lesson on XCM "in the field" and how to use it in the parachain context with examples from how OAK is implementing it.
   - Hands-on content to have students compose XCM and diagnose them (extension of existing content, replacing the assigment we had in PBA2)
1. **TBD - Parachain partners?**
   - Many have expressed interest in this kind of lesson and/or workshop from PBA partners in the ecosystem.

### TBD

1. **Sergei Shulepov (Parity)** - @pepyakin (UCB req. dates: Polkadot mod and maybe more TBD)
   - ~~See #610 for content addition he will lead~~
   - Perhaps a lesson related to pallet-contracts and Wasm if time allows in another mod or as an elective after-class.
   - Perhaps a lesson on XCMP implementation for Polkadot mod (least likely)  in another mod or as an elective after-class.

---

## Additional Content in Consideration

These are to be decided based on being reviewed and approved, then as time available to place them effectively, and deliver at the level of academic rigor and excellence the PBA demands.

1. **HAPPENING :heavy_check_mark: **  Light Clients Lesson (& Exercise) (remote content presentation)
   - **Nikos Kontakis (Parity)** - @wirednkod
   - #544
   - We will identify the right placement for this, likely after we have covered Substrate at least... TBD.
1. **NOT HAPPENING :heavy_minus_sign: ** Applied ZKP short course
   - _No Maths!_ - this would focus on understanding the bleeding edge innovations in the context of Web3 engineering and applications, not the cryptography underpinning it.
   - UCB is the world's premier institution on the subject - thus we are consulting with RDI and looking locally for other SMEs to expand `exotic primitives` to speak to _applied_ ZK tech
   - In consideration: Thor Kamphefner if he is willing and able, and can scale down maths. Example of his great educational work on crypto and ZKPs: https://www.youtube.com/watch?v=zYygxkeGlAA && https://www.youtube.com/live/YwSGyNr_yUU?feature=share&t=6919 && https://uncloak.org/ (a whole course & community on cryptography, too low level for PBA, TBH)
1.  **HAPPENING :heavy_check_mark: ** W3F's researcher Bhagav (@bhargavbh): [Formal Verification short course proposal](https://docs.google.com/document/d/135GtWOQ1OuytaXfbchrHPAgBhiYwv6lY17JcY70VTSY/)
   - #649 

1. **HAPPENING :heavy_check_mark: **  Parity's Security team lead (Vince - @VinceCorsica ): Applied Security short course proposal (remote content presentation)
   - [Draft outline](https://docs.google.com/document/d/1UmNfY2TejqwDMCFFS6CIJqYtuLF4fzk1JG2-DH02Bps/edit) of what will be a proposal issue / PR

**Comment by shawntabrizi** [comment]
Thanks Dan, this helps a lot.

I will start mapping out a syllabus for Polkadot, at least the topics i think should be covered, and look for feedback to fill in the rest.

**Comment by rphmeier** [comment]
BTW, Gabe from InvArch (@arrudagates) is a pro FRAME/Substrate/XCM dev and expressed interest in joining the academy as a teacher/TA but told me he hasn't heard back yet. Very worth considering and reaching out!

**Comment by JoshOrndorff** [comment]
I had offered to Nuke privately to be responsible for the technical aspects of the parachain protocol. @shawntabrizi It seems you are assigned to the Polkadot module, so I'll extend the same offer to you. If you'd like me to own the prep and delivery of the parachains protocol, I'd be excited to do it. And if not, no problem at all. And I agree with what you wrote in your tracking issue, that most of the content about the parachains protocol already exists in slides and videos from the first two waves. I think I could package it up into understandable and effective lessons, exercises, etc

(Not offering to cover staking, governance, fellowship, etc, just how the relay chain validates and process parachain blocks.)

**Reply by shawntabrizi** [reply]
@JoshOrndorff that sounds perfect. I want to cover pretty much all the things you listed you didn't want to cover.

I'd be very happy for you to create and present the parachain protocol and i think between us we should have all the main topics covered.

**Comment by wirednkod** [comment]
@NukeManDan Is there a consideration of an extra Zombienet presentation?

**Reply by nuke-web3** [reply]
Outside of existing content on it?
https://github.com/Polkadot-Blockchain-Academy/pba-content/blob/b0c25efdddc1bd4996232411f951cbcbd186d948/syllabus/6-Polkadot/Zombienet_slides.md

 Sure could be! This also something you wanted to tackle? If so, let's create a separate issue on it to scope that.

**Reply by wirednkod** [reply]
I think Javier is the best, and could also do an online presentation - but I could also do that

> let's create a separate issue on it to scope that.

☝🏽  Lets do that for now

**Comment by nuke-web3** [comment]
FYI folks: I have done a somewhat major update to include dates for people and modules. I will continue to update as things solidify.

There is _some_ flexibility in these dates if we need it, but we need to act _now_ to tweak as we have already starting to book travel with everyone. Please connect with @Sam-Srey-Parity direcly at https://matrix.to/#/@samarith.srey:parity.io to get dates of travel defined.
